cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/env python3
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Cylc configuration linter.
+
+Checks code style, deprecated syntax and other issues."""
+# NOTE: docstring needed for `cylc help all` output
+# (if editing check this still comes out as expected)
+
+COP_DOC = """cylc lint [OPTIONS] ARGS
+
+Cylc lint looks through one or more folders for ".cylc" and ".rc"
+files.
+
+Checks code style, deprecated syntax and other issues.
+
+Can be run either as a linter or "in place" ("-i"), leaving comments
+in files.
```

Comment:
```suggestion
By default, suggestions are written to stdout.
```
In-place mode is covered in the next sentence.
Also I don't think "run as a linter" necessarily implies stdout.
(And many Cylc users won't even know what a linter is.)